File: AboutBoxDlg/dlgabout.cpp
#include "dlgabout.h"

#include "main.h"
#include "appversion.h"

#include <wx/bitmap.h>
#include <wx/statbmp.h>
#include <wx/statline.h>

#include "aboutlogo.h"

DlgAbout::DlgAbout(wxWindow *parent) : wxDialog(parent, -1, wxGetStockLabel(wxID_ABOUT, wxSTOCK_NOFLAGS))
{
	wxStaticBitmap *BmpCtrl;
	wxStaticText *label;
	wxStaticLine *hline;
	wxButton *btnClose;

	wxString sMsg=_T("Made with");
	sMsg << _T(" ") << wxGetApp().GetBuildInfos(false);

	wxString sTitle=_T(PRODUCTNAME);
	sTitle << _T(" (v") << VERSION_MAJOR << _T(".") << VERSION_MINOR << _T(".") << VERSION_REV << _T(".") << VERSION_BUILD << _T(")");

	wxBoxSizer *szrMain=new wxBoxSizer(wxVERTICAL);
		wxBoxSizer *szrTop=new wxBoxSizer(wxHORIZONTAL);
			BmpCtrl=new wxStaticBitmap(this, -1, wxGet_wxWidgets_png_Bitmap(), wxPoint(10,15));
			szrTop->Add(BmpCtrl, 0, wxALL|wxALIGN_CENTER_VERTICAL, 5);
			wxBoxSizer *szrRight=new wxBoxSizer(wxVERTICAL);
				label=new wxStaticText(this, -1, sTitle);
				wxFont fntTitle=label->GetFont();
				fntTitle.MakeLarger();
				fntTitle.MakeBold();
				label->SetFont(fntTitle);
				szrRight->Add(label, 0, wxALL|wxALIGN_CENTER, 5);
				label=new wxStaticText(this, -1, _T(COPYRIGHT));
				szrRight->Add(label, 0, wxLEFT|wxRIGHT|wxBOTTOM|wxALIGN_CENTER, 5);
				label=new wxStaticText(this, -1,_(DESCRIPTION));
				szrRight->Add(label, 0, wxLEFT|wxRIGHT|wxTOP|wxALIGN_CENTER, 5);
				label=new wxStaticText(this, -1, sMsg);
				szrRight->Add(label, 0, wxALL|wxALIGN_CENTER, 5);
			szrTop->Add(szrRight, 0, wxALL|wxALIGN_CENTER_VERTICAL, 5);
		szrMain->Add(szrTop, 0, wxALL, 5);
		hline=new wxStaticLine(this, -1);
		szrMain->Add(hline, 0, wxALL|wxEXPAND, 5);
		btnClose=new wxButton(this, wxID_CANCEL, wxGetStockLabel(wxID_CLOSE));
		szrMain->Add(btnClose, 0, wxALL|wxALIGN_RIGHT, 5);
	SetSizer(szrMain);
	szrMain->SetSizeHints(this);

	CenterOnParent();
}

DlgAbout::~DlgAbout()
{
}

File: AboutBoxDlg/Python/DlgAbout.py
import wx

class DlgAbout(wx.Dialog):
    def __init__(self, parent):
        wx.Dialog.__init__(self, parent, title=wx.GetStockLabel(wx.ID_ABOUT, wx.STOCK_NOFLAGS))

        self._createInterface()

        self.CenterOnParent()

    def _createInterface(self):

        sTitle = 'AboutBoxDlg (v1.0.0)'
        sVersion = wx.version()
        pos = sVersion.find(' wxWidgets')
        if pos != -1:
            sVersion = sVersion[:pos]
        sMsg = 'Made with wxPython ' + sVersion + '\nBased on ' + wx.GetLibraryVersionInfo().VersionString
        szrMain = wx.BoxSizer(wx.VERTICAL)

        szrTop = wx.BoxSizer(wx.HORIZONTAL)

        bmpCtrl = wx.StaticBitmap(self, wx.ID_ANY, wx.Bitmap('../../common/wxWidgets.png', wx.BITMAP_TYPE_PNG))
        szrTop.Add(bmpCtrl, 0, wx.ALL|wx.ALIGN_CENTER_VERTICAL, 5)

        szrRight = wx.BoxSizer(wx.VERTICAL)

        label = wx.StaticText(self, wx.ID_STATIC, sTitle)
        fntTitle = label.GetFont()
        fntTitle.MakeLarger()
        fntTitle.MakeBold()
        label.SetFont(fntTitle)
        szrRight.Add(label, 0, wx.ALL|wx.ALIGN_CENTER, 5)

        label = wx.StaticText(self, wx.ID_STATIC, 'Copyright (c) X.P. 2018')
        szrRight.Add(label, 0, wx.LEFT|wx.RIGHT|wx.BOTTOM|wx.ALIGN_CENTER, 5)

        label = wx.StaticText(self, wx.ID_STATIC, 'wxWidgets Application')
        szrRight.Add(label, 0, wx.LEFT|wx.RIGHT|wx.TOP|wx.ALIGN_CENTER, 5)

        label = wx.StaticText(self, wx.ID_STATIC, sMsg, style=wx.ST_NO_AUTORESIZE|wx.ALIGN_CENTER)
        szrRight.Add(label, 0, wx.ALL|wx.EXPAND, 5)

        szrTop.Add(szrRight, 0, wx.ALL|wx.ALIGN_CENTER_VERTICAL, 5)

        szrMain.Add(szrTop, 0, wx.ALL, 5)

        btnSzr = self.CreateSeparatedButtonSizer(wx.CLOSE)
        szrMain.Add(btnSzr, 0, wx.ALL|wx.EXPAND, 5)

        self.SetSizer(szrMain)

        szrMain.SetSizeHints(self)
